src/ol/renderer/canvas/VectorLayer.js

```diff
@@ -423,13 +423,14 @@ class CanvasVectorLayerRenderer extends CanvasLayerRenderer {
 
       /**
        * @param {import("../../Feature.js").FeatureLike} feature Feature.
+       * @param {import("../../geom/SimpleGeometry.js").default} geometry Geometry.
        * @return {?} Callback result.
        */
-      const featureCallback = function (feature) {
+      const featureCallback = function (feature, geometry) {
         const key = getUid(feature);
         if (!(key in features)) {
           features[key] = true;
-          return callback(feature, layer);
```

Comment:
The `forEachFeatureAtPixel` callback will only be called with the first detected geometry, but multiple geometries of the same feature could be at that pixel.
This should probably also be documented.